### DISCUSSION TITLE
Text tokenizer paragraph support

### DISCUSSION THREAD

**Discussion by JacobDaneh** [readium/kotlin-toolkit#397] [Q&A]
Hi,
I have a customized TTS engine that performs better with paragraphs than sentences.
In swift-toolkit, PublicationSpeechSynthesizer class we can set paragraph for unit parameter of text tokenizer, but unfortunately kotlin-toolkit does not support paragraph. Just word and sentence.

How could I add paragraph as text tokenizer unit?

Thank you for your help

<img width="811" alt="Screenshot 1402-07-03 at 10 31 53" src="https://github.com/readium/kotlin-toolkit/assets/89334459/363d1f60-9206-4ac3-b95e-299fd2fd0c35">
<img width="892" alt="Screenshot 1402-07-03 at 10 33 12" src="https://github.com/readium/kotlin-toolkit/assets/89334459/61caf766-7b20-4238-a856-42dd071b4209">

**Comment by mickael-menu** [comment]
Unfortunately the native ICU `BreakIterator` we use on Android doesn't support paragraph splitting:

https://github.com/readium/kotlin-toolkit/blob/f87dd95afd53e50bfa52e1d59dd4785fc6e4f77e/readium/shared/src/main/java/org/readium/r2/shared/util/tokenizer/TextTokenizer.kt#L57

You could try implementing your own version of a text tokenizer to split by paragraph. Also, if you don't use a text tokenizer at all (or an empty one that doesn't split the input), you will get something that is similar to paragraph tokens, because the HTML content iterator is returning block elements by default.

Let us know if you find a satisfying solution.
